File: haskell-solutions/02.hs
fibonnaci :: Integral a => a -> a
fibonnaci n = round((φ^n-ψ^n)/ sqrt 5)
    where φ = (1+sqrt 5)/2
          ψ = 1-φ

main :: IO ()
main = do
    print $ sum $ filter even $ takeWhile (<4000000) $ map fibonnaci [1..]

File: rust-solutions/02.rs
fn main() {
    let mut sum = 0;
    let mut f1 = 1;
    let mut f2 = 1;
    let mut temp;
    while f2 <= 4000000 {
        if f2%2==0 {
            sum += f2;
        }
        temp = f1 + f2;
        f1 = f2;
        f2 = temp;
    }
    println!("{}", sum);
}